File: mNoisi/Breath/MNTimerViewController.m
//
//  MNTimerViewController.m
//  mNoisi
//
//  Created by yan on 2017/07/12.
//  Copyright © 2017 Leon.yan. All rights reserved.
//

#import "MNTimerViewController.h"
#import "MNTimer.h"

@interface MNTimerViewController () <UIPickerViewDataSource, UIPickerViewDelegate>

@property (nonatomic, weak) IBOutlet UIPickerView *timePicker;

@end

@implementation MNTimerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    if (!_times) {
        _times = @[ @5, @10, @15, @20, @30, @45, @60 ];
    }

    if (_currentTime) {
        NSInteger index = [_times indexOfObject:_currentTime];
        [_timePicker selectRow:index inComponent:0 animated:NO];
    }
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)back:(id)sender
{
    [self.navigationController popViewControllerAnimated:YES];
}

- (IBAction)comfirm:(id)sender
{
    NSInteger row = [_timePicker selectedRowInComponent:0];
    NSInteger time = [_times[row] integerValue];
    if ([_delegate respondsToSelector:@selector(timerViewController:didChooseTime:)]) {
        [_delegate timerViewController:self didChooseTime:time];
    }
    [self.navigationController popViewControllerAnimated:YES];
}

#pragma mark -  UIPickerView
- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView
{
    return 1;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component
{
    return _times.count;
}

- (CGFloat)pickerView:(UIPickerView *)pickerView rowHeightForComponent:(NSInteger)component
{
    return 60;
}

- (NSAttributedString *)pickerView:(UIPickerView *)pickerView attributedTitleForRow:(NSInteger)row forComponent:(NSInteger)component
{
    int value = (int) [_times[row] integerValue];
    NSString *title = [NSString stringWithFormat:@"%d M", value];
    NSDictionary *attributes = @{ NSForegroundColorAttributeName : UIColor.whiteColor};
    return [[NSAttributedString alloc] initWithString:title attributes:attributes];
    
}


@end

File: mNoisi/Breath/MNTimerViewController.swift
//
//  MNTimerViewController.swift
//  mNoisi
//
//  Created by Leon.yan on 25/05/2017.
//  Copyright © 2017 Leon.yan. All rights reserved.
//

import UIKit

protocol MNTimerDelegate: class {

    func timerViewController(_ controller: MNTimerViewController, didChooseTime: Int)
}

class MNTimerViewController: UIViewController, UIPickerViewDataSource, UIPickerViewDelegate {

    weak var delegate: MNTimerDelegate?

    @IBOutlet
    private weak var timePicker: UIPickerView!

    var times: [Int] = [
        1, 3, 5, 10, 15, 20, 25, 30, 40, 50, 60
    ]

    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func back(_ sender: Any) {
        self.navigationController?.popViewController(animated: true)
    }
    
    @IBAction func confirm(_ sender: Any) {
        let selectedRow = timePicker.selectedRow(inComponent: 0)
        self.delegate?.timerViewController(self, didChooseTime: times[selectedRow])
        self.navigationController?.popViewController(animated: true)
    }

    // MARK: UIPickerViewDataSource
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 1
    }

    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return times.count
    }

    func pickerView(_ pickerView: UIPickerView, rowHeightForComponent component: Int) -> CGFloat {
        return 60
    }

    func pickerView(_ pickerView: UIPickerView, attributedTitleForRow row: Int, forComponent component: Int) -> NSAttributedString? {
        let title = "\(times[row]) m"
        return NSAttributedString(string: title, attributes: [ NSForegroundColorAttributeName : UIColor.white ])
    }


    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
